charms/osm/sshproxy.py

```diff
@@ -133,45 +285,78 @@ def run(self, cmd: str) -> (str, str):
 
         # Make sure we have everything we need to connect
         if host and user:
-            return self._ssh(cmd)
+            return self.ssh(cmd)
 
         raise Exception("Invalid SSH credentials.")
 
-    def sftp(self, local, remote):
-        client = self._get_ssh_client()
+    def scp(self, source_file, destination_file):
+        """Execute an scp command. Requires a fully qualified source and
+        destination.
+
+        :param str source_file: Path to the source file
+        :param str destination_file: Path to the destination file
+        :raises: :class:`CalledProcessError` if the command fails
+        """
+        cmd = [
+            "sshpass",
+            "-p",
+            self.password,
+            "scp",
+            "-i",
+            os.path.expanduser(self.private_key_path),
+            "-o",
+            "StrictHostKeyChecking=no",
+            "-q",
+            "-B",
+        ]
+        destination = "{}@{}:{}".format(self.username, self.hostname, destination_file)
+        cmd.extend([source_file, destination])
+        subprocess.run(cmd, check=True)
+
+    def ssh(self, command):
+        """Run a command remotely via SSH.
+
+        :param list(str) command: The command to execute
+        :return: tuple: The stdout and stderr of the command execution
+        :raises: :class:`CalledProcessError` if the command fails
+        """
 
-        # Create an sftp connection from the underlying transport
-        sftp = paramiko.SFTPClient.from_transport(client.get_transport())
-        sftp.put(local, remote)
-        client.close()
-        pass
+        destination = "{}@{}".format(self.username, self.hostname)
+        cmd = [
+            "sshpass",
+            "-p",
+            self.password,
+            "ssh",
+            "-i",
+            os.path.expanduser(self.private_key_path),
+            "-o",
+            "StrictHostKeyChecking=no",
+            "-q",
+            destination,
+        ]
+        cmd.extend(command)
+        output = subprocess.run(cmd, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        return (output.stdout.decode("utf-8").strip(), output.stderr.decode("utf-8").strip())
 
     def verify_credentials(self):
         """Verify the SSH credentials.
         
         :return (bool, str): Verified, Stderr
         """
+        verified = True
```

Comment:
why not make this `verified = False` and add a `verified = True` after the run command?